lib/web/apiserver.go

```diff
@@ -2652,7 +2652,11 @@ type ssoRequestParams struct {
 }
 
 func parseSSORequestParams(r *http.Request) (*ssoRequestParams, error) {
-	query := r.URL.Query()
+	rawQuery := strings.Replace(r.URL.RawQuery, "+", "%2B", -1)
+	query, err := url.ParseQuery(rawQuery)
```

Comment:
I think it's OK to ignore this error here and proceed with the empty string instead. This is what `query := r.URL.Query()` does too. 

```golang
query, _ := url.ParseQuery(rawQuery)
```